kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/dek/BufferTooSmallException.java

```diff
@@ -10,4 +10,9 @@ public class BufferTooSmallException extends RuntimeException {
     public BufferTooSmallException() {
         super();
     }
+
+    @Override
+    public synchronized Throwable fillInStackTrace() {
+        return null;
```

Comment:
Doesn't returning null break the contract?  Throwable#fillInStackTrace is documented as "returns
a reference to this Throwable instance." so by returning null you break that contract.

Reply:
How about using a different super [constructor](https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#Throwable-java.lang.String-java.lang.Throwable-boolean-boolean-)?  e.g. `super("Buffer was to small", null, true, false);` Where the last boolean tells Throwable not to fill in the stacktrace.

Alternatively personally I would just have a single instance of the buffer to small exception in [WrapperV2](https://github.com/kroxylicious/kroxylicious/blob/574f67316e69831f259df689f12865d922ab2ffb/kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/crypto/WrapperV2.java#L119).

Though like Keith I'm surprised this is showing up in the performance profile.

Reply:
> Alternatively personally I would just have a single instance of the buffer to small exception in [WrapperV2](https://github.com/kroxylicious/kroxylicious/blob/574f67316e69831f259df689f12865d922ab2ffb/kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/crypto/WrapperV2.java#L119).

Yeah, if it is truly hot, that's the way I go.  By my reading of the code, it is not a hot path.

Reply:
It came up while running the performance job on my machine in the flame graphs, but I'm still investigating

Reply:
thanks @lucamolteni if its showing up as meaningful then I think we have two things to fix.

1. One of the options mentioned here to reduce the costs.
2. Work out why its showing up as it really should be an exceptional case. 

Though given the stacktrace isn't really going to help anyone understand why the buffer was too small I'm tempted to just make one of the fixes anyway.

Reply:
Just clearing my admin backlog did we get any where figuring out why this was showing up under profiling?

Reply:
Sorry Sam I got sidetracked by another project, it definetly appeared inside the profiling data, but it doesn't seem to have that much of an impact in the performance after all. 

I agree that the stack trace is not useful at all here, and more generally that using exceptions for the control flow seems to be risky for the future, so I'd merge it anyway, but either way is fine.

Reply:
>  it definitely appeared inside the profiling data, 

Mmm, that's fishy. Do you have a reproduction method that will cause it?